schema-registry/serdes/src/main/java/com/hortonworks/registries/schemaregistry/serdes/avro/SchemaVersionProtocolHandlerRegistry.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2016 Hortonworks.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hortonworks.registries.schemaregistry.serdes.avro;
+
+import com.hortonworks.registries.schemaregistry.SchemaIdVersion;
+import com.hortonworks.registries.schemaregistry.serde.SerDesException;
+import com.hortonworks.registries.schemaregistry.serdes.AbstractSchemaVersionProtocolHandler;
+import com.hortonworks.registries.schemaregistry.serdes.SchemaVersionProtocolHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Registered ser/des protocol handlers which can be used in serializers/deserializers.
+ */
+public final class SchemaVersionProtocolHandlerRegistry {
+    public static final byte METADATA_ID_VERSION_PROTOCOL = 0x1;
+    public static final byte VERSION_ID_AS_LONG_PROTOCOL = 0x2;
+    public static final byte VERSION_ID_AS_INT_PROTOCOL = 0x3;
```

Comment:
I do not want to mix our protocol with confluent's. We can continue to have our protocol with version identifier as int and our serialized data format. That is why it is not added as part of protocol version 0x0, which should be according to Confluent's serialization.